tests/IteratorTest.php

```diff
@@ -227,6 +228,19 @@ public function testOffsetGet()
         $this->assertEquals(1, self::$iterator[1]->id());
     }
 
+    public function testToArray()
+    {
+        $query = new Query(IteratorTestModel::class);
+        $query->limit(3);
+        $iterator = new Iterator($query);
+        $iterator->next();
```

Comment:
Is it a requirement for next() to be called first in order for toArray() to work?

Reply:
no, we just check that array order is not broken, because of iteration